source/beamline/CLS/CLSShutters.cpp

```diff
@@ -14,7 +19,88 @@ CLSShutters::CLSShutters(const QString &name, QObject *parent) :
 
 CLSShutters::~CLSShutters()
 {
+	shuttersBeamOnOrderMap_.clear();
+}
+
+AMListAction3* CLSShutters::createBeamOnActionList()
```

Comment:
it is just opening/closing the shutters/valves. and yes, it will be better to reimplement the `AMPseudoMotorControl::createMoveAction(double)`